src/object/mod.rs

```diff
@@ -2,70 +2,191 @@
 
 pub(crate) mod class;
 
-use neon_runtime;
 use neon_runtime::raw;
-use handle::{Handle, Managed};
-use types::{Value, JsValue, JsArray, build};
-use types::utf8::Utf8;
-use context::Context;
-use result::{NeonResult, JsResult, Throw};
+use handle::Managed;
 
 pub use self::class::{Class, ClassDescriptor};
+pub use self::internal::*;
 
-/// A property key in a JavaScript object.
-pub trait PropertyKey {
-    unsafe fn get_from(self, out: &mut raw::Local, obj: raw::Local) -> bool;
-    unsafe fn set_from(self, out: &mut bool, obj: raw::Local, val: raw::Local) -> bool;
-}
+#[cfg(feature = "legacy-runtime")]
+mod internal {
```

Comment:
I'm not super keen on this, but it seemed slightly better than adding an attribute to _everything_. Possible alternatives:

* `cfg-if`. This reduces some boilerplate and allows shared `use` lines, but also adds an extra level of indentation.
* Move the implementation to separate files. `internal_legacy.rs` and `internal_napi.rs`.

The name `internal` is inconsistent with other uses of `internal` which imply it's private. Do you have other name suggestions?

Reply:
separate files would make sense I think, given that almost the entire module is wrapped in feature gates. if keeping them in one file `mod implementation` would work a bit better than `mod internal`, imho